src/lib/components/EventSchedule/index.js

```diff
@@ -177,15 +178,25 @@ class EventSchedule extends HTMLElement {
     this._tabsElement = null;
   }
 
-  onStateChanged({activeEventDay}) {
+  onStateChanged({activeEventDay, isModalOpen}) {
     this._activeEventDay = activeEventDay;
 
-    // This relies on the event data being in the same shape as the rendered
-    // tabs, which is pretty safe, since it comes from the same source.
-    // Don't change the tab for the user if a modal is already open.
-    if (!this._modalElement.open && activeEventDay) {
-      this._tabsElement.activeTab = activeEventDay.index;
+    // If the user has clicked to a different day, opened and closed a modal
+    // then don't do anything.
+    // If this is not the case then it means that either we're loading the
+    // page for the first time, or the user has left the tab open and we're
+    // transitioning to the next day. In that case, update the tabs element
+    // to reflect the current day.
+    if (!this._wasModalOpen) {
```

Comment:
Can't this all be merged into one `if` satetement?

Reply:
mmmmm yep. I think so. let me do that.